backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceMetrics.java

```diff
@@ -0,0 +1,309 @@
+        this.context = context;
+        this.customAttributes = customAttributes;
+        this.settings = settings;
+    }
+
+    @Override
+    public void startMetricsEventHandlers(Api backtraceApi) {
+        uniqueEventsHandler = backtraceApi.enableUniqueEvents(context, customAttributes, settings);
+        summedEventsHandler = backtraceApi.enableSummedEvents(context, customAttributes, settings);
+    }
+
+    void setStartupUniqueEventName(String StartupUniqueEventName) {
+        this.startupUniqueEventName = StartupUniqueEventName;
+    }
+
+    String getStartupUniqueEventName() {
+        return this.startupUniqueEventName;
+    }
+
+    void setMaximumEvents(int MaximumEvents) {
+        this.maximumEvents = MaximumEvents;
+    }
+
+    int getMaximumEvents() {
+        return this.maximumEvents;
+    }
+
+    public String getBaseUrl() {
+        return this.settings.getBaseUrl();
+    }
+
+    @SuppressWarnings("unchecked, rawtypes")
+    public ConcurrentLinkedDeque<UniqueEvent> getUniqueEvents() {
+        for (Event event : uniqueEventsHandler.events) {
+            if (!(event instanceof UniqueEvent)) {
+                BacktraceLogger.e(LOG_TAG, "UniqueEventsHandler contains an event of type " + event.getClass() + " this is a bug");
+                return new ConcurrentLinkedDeque<>();
+            }
+        }
+        return (ConcurrentLinkedDeque) this.uniqueEventsHandler.events;
+    }
+
+    @SuppressWarnings("unchecked, rawtypes")
+    public ConcurrentLinkedDeque<SummedEvent> getSummedEvents() {
+        for (Event event : summedEventsHandler.events) {
+            if (!(event instanceof SummedEvent)) {
+                BacktraceLogger.e(LOG_TAG, "SummedEventsHandler contains an event of type " + event.getClass() + " this is a bug");
+                return new ConcurrentLinkedDeque<>();
+            }
+        }
+        return (ConcurrentLinkedDeque) this.summedEventsHandler.events;
+    }
+
+    /**
+     * Send startup event to Backtrace
+     */
+    public void sendStartupEvent() {
+        uniqueEventsHandler.sendStartupEvent(startupUniqueEventName);
+        summedEventsHandler.sendStartupEvent(startupSummedEventName);
+    }
+
+    /**
+     * Send all outgoing messages (unique and summed) currently queued
+     */
+    public void send() {
+        uniqueEventsHandler.send();
+        summedEventsHandler.send();
+    }
+
+    /**
+     * Add a unique event to the next Backtrace Metrics request
+     * @param attributeName Attribute name
+     * @return true if success
+     */
+    public boolean addUniqueEvent(String attributeName) {
+        return addUniqueEvent(attributeName, null);
+    }
+
+    /**
+     * Add a unique event to the next Backtrace Metrics request
+     * @param attributeName Attribute name
+     * @param attributes    Event attributes
+     * @return true if success
+     */
+    public boolean addUniqueEvent(String attributeName, Map<String, Object> attributes) {
```

Comment:
attributeName is a typo?

Reply:
I guess this is the point to which attribute is the unique attribute - call it uniqueAttributeName or attributeNameForUniqueness or smth?

Reply:
I just followed `backtrace-unity`:
https://github.com/backtrace-labs/backtrace-unity/blob/553aab2b39c318ff96ebed4bc739bf2c87304649/Runtime/Services/BacktraceMetrics.cs#L297

Reply:
Should we still change it @vlussenburg @konraddysput ?

Reply:
I will mark this as resolved @vlussenburg @konraddysput